### ISSUE TITLE
When installing a chart app without an icon the default icon svg flashes as we GET it numerous times while provisioning

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#17668]
Version: master 2.2 (1/28/19)

**What kind of request is this (question/bug/enhancement/feature request):**
Bug, Minor

**Steps to reproduce (least amount of steps as possible):**
- Create either a standard or MC app with any app that does not have an icon such as docker-registry.
- Watch the app as it is provisioned and notice that it flashes on the screen.

**Result:**
The default svg icon flashes on the screen. If you inspect the HTTP we are GET'ing the icon svg multiple times as a text/plain MIME type.
![image](https://user-images.githubusercontent.com/45179589/51863330-f7255500-22fd-11e9-94cc-fc50a9c47243.png)

**Other details that may be helpful:**
Does not happen if the app has an icon.

**Comment by westlywright** [comment]
I ran into this while working on another issue. Will be addressed by rancher/rancher#2897

**Comment by davidnuzik** [comment]
The svg icon still flashes for me, but perhaps not quite as often. We still try to get the icon multiple times.

To reproduce:
Add my catalog: 

https://github.com/davidnuzik/charts.git
(master branch)

Try to install the simplenginx app. Quickly view the Apps page - inspect HTTP. You'll see we try to get the icon several times and it will flash on the screen.

**Comment by davidnuzik** [comment]
Version: Master (v2.3) (5/20/19)

The icon still flashes when the state changes (I.e. deploying then to Active, etc) but this is significantly improved. Based on a discussion with Wes this is what we are going with for now and this is fine with me. I only saw 3 additional times we GET the icon svg (and it flashes) again for each time the state is changed for the app. This is much, much better than compared to before when we did this many times as seen in the original issue description screenshot. The flashing of the icon is extremely brief and not a big issue so I am not worried about it at this time.

![image](https://user-images.githubusercontent.com/45179589/58037502-6956bb00-7ae2-11e9-9a6b-030d7ac29dca.png)
